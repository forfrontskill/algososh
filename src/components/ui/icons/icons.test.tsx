import renderer from 'react-test-renderer';
import { ArrowIcon } from './arrow-icon';
import { AscendingIcon } from './ascending-icon';
import { DescendingIcon } from './descending-icon';
import { ReturnIcon } from './return-icon';

describe('Test Icons components', () => {

    it('ArrowIcon rendering', () => {
        const tree = renderer
            .create(<ArrowIcon />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('AscendingIcon rendering', () => {
        const tree = renderer
            .create(<AscendingIcon />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('DescendingIcon rendering', () => {
        const tree = renderer
            .create(<DescendingIcon />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('ReturnIcon rendering', () => {
        const tree = renderer
            .create(<ReturnIcon />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});