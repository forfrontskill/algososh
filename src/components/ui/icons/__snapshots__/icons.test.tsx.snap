// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Icons components ArrowIcon rendering 1`] = `
<svg
  fill="none"
  height="24"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    clipRule="evenodd"
    d="M9.29289 4.29289C9.68342 3.90237 10.3166 3.90237 10.7071 4.29289L18.4142 12L10.7071 19.7071C10.3166 20.0976 9.68342 20.0976 9.29289 19.7071C8.90237 19.3166 8.90237 18.6834 9.29289 18.2929L15.5858 12L9.29289 5.70711C8.90237 5.31658 8.90237 4.68342 9.29289 4.29289Z"
    fill="#0032FF"
    fillRule="evenodd"
  />
</svg>
`;

exports[`Test Icons components AscendingIcon rendering 1`] = `
<svg
  fill="inherit"
  height="12"
  viewBox="0 0 20 12"
  width="20"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M1 0C0.447715 0 0 0.447715 0 1C0 1.55228 0.447715 2 1 2H11C11.5523 2 12 1.55228 12 1C12 0.447715 11.5523 0 11 0H1ZM0 6C0 5.44772 0.447715 5 1 5H15C15.5523 5 16 5.44772 16 6C16 6.55228 15.5523 7 15 7H1C0.447715 7 0 6.55228 0 6ZM0 11C0 10.4477 0.447715 10 1 10H19C19.5523 10 20 10.4477 20 11C20 11.5523 19.5523 12 19 12H1C0.447716 12 0 11.5523 0 11Z"
    fill="inherit"
    fillOpacity="0.85"
  />
</svg>
`;

exports[`Test Icons components DescendingIcon rendering 1`] = `
<svg
  fill="inherit"
  height="12"
  viewBox="0 0 20 12"
  width="20"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    clipRule="evenodd"
    d="M0 1C0 0.44772 0.44772 0 1 0H19C19.5523 0 20 0.44772 20 1C20 1.55228 19.5523 2 19 2H1C0.44772 2 0 1.55228 0 1ZM0 6C0 5.4477 0.44772 5 1 5H15C15.5523 5 16 5.4477 16 6C16 6.5523 15.5523 7 15 7H1C0.44772 7 0 6.5523 0 6ZM1 10C0.44772 10 0 10.4477 0 11C0 11.5523 0.44772 12 1 12H11C11.5523 12 12 11.5523 12 11C12 10.4477 11.5523 10 11 10H1Z"
    fill="inherit"
    fillOpacity="0.85"
    fillRule="evenodd"
  />
</svg>
`;

exports[`Test Icons components ReturnIcon rendering 1`] = `
<svg
  fill="none"
  height="24"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    clipRule="evenodd"
    d="M12.7073 4.29289C13.0978 4.68342 13.0978 5.31658 12.7073 5.70711L7.41436 11H19.0002C19.5524 11 20.0002 11.4477 20.0002 12C20.0002 12.5523 19.5524 13 19.0002 13H7.41436L12.7073 18.2929C13.0978 18.6834 13.0978 19.3166 12.7073 19.7071C12.3167 20.0976 11.6836 20.0976 11.293 19.7071L3.58594 12L11.293 4.29289C11.6836 3.90237 12.3167 3.90237 12.7073 4.29289Z"
    fill="#0032FF"
    fillRule="evenodd"
  />
</svg>
`;
